cassiopeia/tools/coupling.py

```diff
@@ -0,0 +1,125 @@
+"""
+File storing functionality for computing coupling statistics between meta
+variables on a tree.
+"""
+from typing import Callable, Optional
+
+from collections import defaultdict
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.data import utilities as data_utilities
+
+
+def compute_evolutionary_coupling(
+    tree: CassiopeiaTree,
+    meta_variable: str,
+    minimum_proportion: float = 0.05,
+    number_of_shuffles: int = 500,
+    random_state: Optional[np.random.RandomState] = None,
+    dissimilarity_map: Optional[pd.DataFrame] = None,
+    cluster_comparison_function: Optional[
+        Callable
+    ] = data_utilities.net_relatedness_index,
```

Comment:
Think this is a typo. `Optional` is only used for arguments that are `None` by default.

Reply:
You're totally right! I'll get rid of this.